src/redux/util.js

```diff
@@ -38,40 +38,75 @@ export function sortByString(list, key) {
   return sortByKey(list, elem => elem[key].toLowerCase());
 }
 
-export function sortByKey(list, key, order='asc') {
+export function sortByKey(list, key, order = 'asc') {
   return _.orderBy(list, [key], [order]);
 }
 
-export async function fetchAllRecordsUsingFetchWithBQ(fetchWithBQ, url, itemsPerPage, otherParams = null) {
-  const fetch = async (page) => {
+export async function fetchAllRecordsUsingFetchWithBQ(
+  fetchWithBQ,
+  url,
+  itemsPerPage,
+  otherParams = null,
+) {
+  const fetch = async page => {
     const params = new URLSearchParams({
       pagination: true,
       page,
       itemsPerPage,
       ...otherParams,
     });
-    const result = await fetchWithBQ(`${url}?${params.toString()}`);
-    return result.data;
 
+    try {
+      const result = await fetchWithBQ(`${url}?${params.toString()}`);
+      return { data: result.data, error: null };
+    } catch (err) {
+      return { data: null, error: err };
+    }
   };
-  const firstRs = await fetch(1);
 
-  if (!Object.hasOwn(firstRs, 'hydra:totalItems') || firstRs['hydra:totalItems'] <= firstRs['hydra:member'].length) {
+  const firstPageResult = await fetch(1);
+  if (firstPageResult.error) {
+    return { error: firstPageResult.error };
```

Comment:
Better to return the whole object: `return firstPageResult`.
We might need/want to access/use other response properties from outside (we don't know at this scope what the "outside-world" would do/need from a failed result, although we are only using this from `src/redux/api/slice.js` right now).